Buildings/Templates/ChilledWaterPlant/Components/PrimaryPumpGroup/BaseClasses/DedicatedPrimaryPumps.mo

```diff
@@ -0,0 +1,128 @@
+    annotation (Placement(transformation(extent={{-108,-30},{-92,30}}),
+        iconTransformation(extent={{-108,-30},{-92,30}})));
+  Modelica.Fluid.Interfaces.FluidPort_b port_b(
+    redeclare each final package Medium = Medium,
+    each m_flow(max=0),
+    each h_outflow(start=Medium.h_default, nominal=Medium.h_default))
+    "Fluid connector b (positive design flow direction is from port_a to port_b)"
+    annotation (Placement(transformation(extent={{110,-10},{90,10}}),
+        iconTransformation(extent={{110,-10},{90,10}})));
+
+  replaceable parameter Fluid.Movers.Data.Generic per(pressure(
+    V_flow=m_flow_nominal/1000 .* {0,1,2},
+    dp=dp_nominal .* {1.5,1,0.5}))
+    constrainedby Fluid.Movers.Data.Generic
+    "Performance data"
+    annotation (
+      choicesAllMatching=true,
+      Placement(transformation(extent={{-90,-88},{-70,-68}})));
+  replaceable Fluid.Movers.SpeedControlled_y pum[nPum](
+    each energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial)
+    constrainedby Fluid.Movers.SpeedControlled_y(
+      redeclare each final package Medium=Medium,
+      each final inputType=Buildings.Fluid.Types.InputType.Continuous,
+      each final per=per) "Pumps"
+    annotation (
+      choicesAllMatching=true,
+      Placement(transformation(extent={{-30,-10},{-10,10}})));
+  Buildings.Controls.OBC.CDL.Continuous.GreaterThreshold evaSta[nPum](
+    t=1E-2, h=0.5E-2) "Evaluate pump status"
+    annotation (Placement(transformation(
+        extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={50,70})));
+  Buildings.Controls.OBC.CDL.Continuous.Hysteresis hys[nPum](
+    each final uLow=threshold, each final uHigh=2*threshold)
+    "Hysteresis for isolation valves"
+    annotation (Placement(transformation(extent={{-60,40},{-40,60}})));
+  Buildings.Controls.OBC.CDL.Conversions.BooleanToReal booToRea[nPum]
+    "Boolean to real conversion for isolation valves"
+    annotation (Placement(transformation(extent={{-20,40},{0,60}})));
+  Fluid.Actuators.Valves.TwoWayLinear val[nPum](
+    redeclare each final replaceable package Medium = Medium,
+    each final dpFixed_nominal=0,
+    each final CvData=Buildings.Fluid.Types.CvTypes.OpPoint,
+    each final dpValve_nominal=dpValve_nominal,
+    each final m_flow_nominal=m_flow_nominal)
+    "Isolation valves"
+    annotation (Placement(transformation(extent={{10,-10},{30,10}})));
+  Modelica.Blocks.Interfaces.RealInput y[nPum](unit="1")
+    annotation (Placement(
+        transformation(
+        extent={{-20,-20},{20,20}},
+        rotation=-90,
+        origin={0,100}),
+        iconTransformation(
+        extent={{-20,-20},{20,20}},
+        rotation=-90,
+        origin={0,120})));
+  Modelica.Blocks.Interfaces.BooleanOutput y_actual[nPum]
+    annotation (Placement(
+        transformation(extent={{100,50},{140,90}}),
+        iconTransformation(extent={{100,70},
+            {120,90}})));
+  Fluid.MixingVolumes.MixingVolume vol(nPorts=1)
```

Comment:
Missing partial medium redeclaration and nominal mass flow rate assignment.
Fix `final nPorts=nPum + 1`.